### DISCUSSION TITLE
Linux install proceeds despite curl errors

### DISCUSSION THREAD

**Discussion by ddelange** [terraform-linters/tflint#1867] [General]
### Summary

`install_linux.sh` fails in Bitbucket Pipelines with an alpine image:
```console
$ apk add curl coreutils bash unzip musl-dev gcc python3-dev py3-pip
$ curl -s https://raw.githubusercontent.com/terraform-linters/tflint/master/install_linux.sh | bash
arch=amd64
os=linux_amd64
====================================================
Looking up the latest version ...
curl: (22) The requested URL returned error: 403
Downloading TFLint 
curl: (22) The requested URL returned error: 404
Downloaded successfully
====================================================
Unpacking /tmp/tflint.GgqLkXBDN1/tflint.zip ...
unzip:  cannot find or open /tmp/tflint.GgqLkXBDN1/tflint.zip, /tmp/tflint.GgqLkXBDN1/tflint.zip.zip or /tmp/tflint.GgqLkXBDN1/tflint.zip.ZIP.
Installing /tmp/tflint.GgqLkXBDN1/tflint to /usr/local/bin/ ...
install: cannot stat '/tmp/tflint.GgqLkXBDN1/tflint': No such file or directory
Failed to install tflint
$ TFLINT_VERSION=v0.48.0 curl -s https://raw.githubusercontent.com/terraform-linters/tflint/v0.48.0/install_linux.sh | bash
arch=amd64
os=linux_amd64
====================================================
Looking up the latest version ...
curl: (22) The requested URL returned error: 403
Downloading TFLint 
curl: (22) The requested URL returned error: 404
Downloaded successfully
====================================================
Unpacking /tmp/tflint.kT03zGhfah/tflint.zip ...
unzip:  cannot find or open /tmp/tflint.kT03zGhfah/tflint.zip, /tmp/tflint.kT03zGhfah/tflint.zip.zip or /tmp/tflint.kT03zGhfah/tflint.zip.ZIP.
Installing /tmp/tflint.kT03zGhfah/tflint to /usr/local/bin/ ...
install: cannot stat '/tmp/tflint.kT03zGhfah/tflint': No such file or directory
Failed to install tflint
```

In the install script, I see curl `--fail` flag:
```
 -f, --fail          Fail silently (no output at all) on HTTP errors
     --fail-early    Fail on first transfer error, do not continue
```

So I think there are two things going on here:
- The script continues with `Downloading TFLint` and `Downloaded successfully` (despite 403 and 404 respectively).
- In the second command, I pass TFLINT_VERSION explicitly, yet still I see `Looking up the latest version ...` -- an api request that apparently gets blocked with 403 for unauthenticated requests coming from Bitbucket Pipelines

### Command

tflint

### Terraform Configuration

```terraform
--
```

### TFLint Configuration

```HCL
--
```

### Output

```Shell
--
```

### TFLint Version

0.48.0 / master

### Terraform Version

--

### Operating System

- [X] Linux
- [ ] macOS
- [ ] Windows

**Comment by bendrucker** [comment]
I closed the PR as it's assuredly not the right change, `--fail` is important. More info is needed here before proceeding with any changes. To the extent there's a bug, it would be that the error returned from cURL doesn't cause the script to exit immediately. Removing `--fail` just obscures existing errors in your environment that must be fixed for the script to work.

**Reply by bendrucker** [reply]
See https://github.com/terraform-linters/tflint/pull/1868 in which I am unable to reproduce any bug. The script could definitely use a touch-up but we'll need a failing, reproducible test case before we make any changes based on your observations.

**Reply by ddelange** [reply]
Hi @bendrucker 👋

To reproduce, you'll need the `curl script | bash` pattern.

Please see my last comments on the https://github.com/terraform-linters/tflint/pull/1866 :)

**Reply by bendrucker** [reply]
Gotcha, so the issue here is not that the script is failing, but that the error doesn't immediately halt it. Whatever's going on with the token you're providing causing those HTTP errors will still require fixing, on your end.
